Tests/VectorUtils.h

```diff
@@ -24,4 +24,21 @@ void fillVectorRandomly(Vector& x)
     for (Index i = 0; i < size; ++i) {
         MathLib::setVector(x, i, rnd(random_number_generator));
     }
+#ifdef USE_PETSC
+    x.finalizeAssembly();
+#endif
+}
+
+template <typename Vector>
```

Comment:
You are right, I changed the implementation of the test and forgot to remove it. I will do it now.